docs/datatypes/tpriorityqueue.md

```diff
@@ -0,0 +1,70 @@
+---
+id: datatypes_tpriorityqueue
+title:  "TPriorityQueue"
+---
+
+A `TPriorityQueue[A]` is a mutable queue that can participate in STM transactions. A `TPriorityQueue` contains values of type `A` that an `Ordering` is defined on. Unlike a `TQueue`, `take` returns the highest priority value (the value that is first in the specified ordering) as opposed to the first value offered to the queue. The ordering that elements with the same priority will be taken from the queue is not guaranteed.
+
+## Creating a TPriorityQueue
+
+You can create an empty `TPriorityQueue` using the `empty` constructor:
+
+```scala mdoc:silent
+import zio._
+import zio.stm._
+
+val minQueue: STM[Nothing, TPriorityQueue[Int]] =
+  TPriorityQueue.empty
+```
+
+Notice that a `TPriorityQueue` is created with an implicit `Ordering`. By default, `take` will return the value that is first in the specified ordering. For example, in a queue of events ordered by time the earliest event would be taken first. If you want a different behavior you can use a custom `Ordering`.
+
+```scala mdoc:silent
+val maxQueue: STM[Nothing, TPriorityQueue[Int]] =
+  TPriorityQueue.empty(Ordering[Int].reverse)
+```
+
+You can also create a `TPriorityQueue` initialized with specified elements using the `fromIterable` or `make` constructors". The `fromIterable` constructor takes a `Iterable` while the `make` constructor takes a variable arguments sequence of elements.
+
+## Offering elements to a TPriorityQueue
+
+You can offer elements to a `TPriorityQueue` using the `offer` or `offerAll` methods. The `offerAll` method is more efficient if you want to offer more than one element to the queue at the same time.
+
+```scala mdoc:silent
+val queue: STM[Nothing, TPriorityQueue[Int]] =
+  for {
+    queue <- TPriorityQueue.empty[Int]
+    _     <- queue.offerAll(List(2, 4, 6, 3, 5, 6))
+  } yield queue
+```
+
+## Taking elements from a TPriorityQueue
+
+Take an element from a `TPriorityQueue` using the `take`. `take` will semantically block until there is at least one value in the queue to take. You can also use `takeAll` to immediately take all values that are currently in the queue, or `takeUpTo` to immediately take up to the specified number of elements from the queue.
+
+```scala mdoc:silent
+val sorted: STM[Nothing, Chunk[Int]] =
+  for {
+    queue  <- TPriorityQueue.empty[Int]
+    _      <- queue.offerAll(List(2, 4, 6, 3, 5, 6))
+    sorted <- queue.takeAll
+  } yield sorted
+```
+
+You can also use `takeOption` method to take the first value from the queue if it exists without suspending or the `poll` method to observe the first element of the queue if it exists without removing it from the queue.
```

Comment:
is not it `peek` rather than `poll`? Peek is the standard term for such an operation.

Reply:
Yes you're right. Good catch!